src/libs/antares/study/parts/hydro/finallevelinflowsmodifyer.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+** Copyright 2007-2023 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include "finallevelinflowsmodifyer.h"
+#include "container.h"
+
+namespace Antares::Data
+{
+
+bool isValidLevel(double level)
+{
+    return level >= 0. && !isnan(level);
+}
+
+FinalLevelInflowsModifier::FinalLevelInflowsModifier(const PartHydro& hydro,
+                                                     const unsigned int& areaIndex,
+                                                     const AreaName& areaName) :
+    hydro_(hydro), areaIndex_(areaIndex), areaName_(areaName)
+{
+}
+
+bool FinalLevelInflowsModifier::CheckInfeasibility(uint year)
+{
+    ComputeDelta(year);
+
+    if (!isActive())
```

Comment:
**Question** : 
Here we have : 
```C++
if (!isActive())
    return true;
```
We could expect to return **false**, not **true**. For instance if the initial level is a **nan**, we cannot exploit it, and so we should prevent any final level computation for this area and this hour.
So why do we return **true** ?

Reply:
Similar to https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1564#discussion_r1317866196
We return true just to skip:
![image](https://github.com/AntaresSimulatorTeam/Antares_Simulator/assets/97689304/43d5efdf-7a9b-414f-9092-220e315b188e)
but not to stop the simulation. 
We return false if pre-checks fail and then we stop the simulation.